test/unit/puppetlabs/services/request_handler/request_handler_core_test.clj

```diff
@@ -326,31 +329,28 @@
            (deliver second-promise true)))))))
 
 (deftest request-handler-test
-  (let [dummy-service (reify jruby-protocol/JRubyPuppetService
-                        (borrow-instance [_ _] {})
-                        (return-instance [_ _ _])
-                        (free-instance-count [_])
-                        (mark-all-environments-expired! [_])
-                        (flush-jruby-pool! [_]))
-        dummy-service-with-timeout (reify jruby-protocol/JRubyPuppetService
-                                     (borrow-instance [_ _] nil)
-                                     (return-instance [_ _ _])
-                                     (free-instance-count [_])
-                                     (mark-all-environments-expired! [_])
-                                     (flush-jruby-pool! [_]))]
-    (logutils/with-test-logging
-      (testing "slingshot bad requests translated to ring response"
-        (let [bad-message "it's real bad"
-              request-handler (core/build-request-handler dummy-service {} (constantly nil))]
-          (with-redefs [core/as-jruby-request (fn [_ _]
-                                                (ringutils/throw-bad-request!
-                                                  bad-message))]
-            (let [response (request-handler {:body (StringReader. "blah")})]
-              (is (= 400 (:status response)) "Unexpected response status")
-              (is (= bad-message (:body response)) "Unexpected response body")))
-          (let [request-handler (core/build-request-handler dummy-service-with-timeout {} (constantly nil))
-                response (request-handler {:body (StringReader. "")})]
-            (is (= 503 (:status response)) "Unexpected response status")
-            (is (.startsWith
-                  (:body response)
-                  "Attempt to borrow a JRuby instance from the pool"))))))))
+    (let [dummy-service (reify jruby-protocol/JRubyPuppetService
+                          (get-pool-context [_]
+                            (jruby-pool-manager-core/create-pool-context
+                             (jruby-core/initialize-config {:gem-home "foo"
+                                                            :ruby-load-path ["bar"]}))))]
+      (logutils/with-test-logging
+       (testing "slingshot bad requests translated to ring response"
+         (let [bad-message "it's real bad"]
+
+           (with-redefs [core/as-jruby-request (fn [_ _]
+                                                 (ringutils/throw-bad-request!
+                                                  bad-message))
+                         jruby-core/return-to-pool (fn [_ _ _] #())]
+             (with-redefs [jruby-core/borrow-from-pool-with-timeout (fn [_ _ _] {})]
```

Comment:
Would it be possible to avoid the `borrow-from-pool-with-timeout` with-redefs if this test were to fully initialize the pool with a just single "non-Puppet" instance in it - with a `jruby-pool-manager-core/create-pool` call instead of just a `jruby-pool-manager-core/create-pool-context` one?